src/restruct_poc/constraints/calculate_prescribed_bc_constraint.hpp

```diff
@@ -76,7 +79,9 @@ struct CalculatePrescribedBCConstraint {
         //----------------------------------------------------------------------
 
         // Extract residual rows relevant to this constraint
-        const auto Phi = Kokkos::subview(Phi_, cd.row_range);
+        const auto Phi = Kokkos::subview(
```

Comment:
This same code is repeated through in 6 files. If it ever needs to be modified then it will need to be updated 6 times. A new developer is almost guaranteed to miss one of the spots where it needs to be updated. This should be moved to the calling function and passed in since all of the calculate constraint functions need it.

Reply:
Similar to above, I'll go back to the row_range being a pair, so the code will look much the same as it did before.

Are you suggesting that I write a function like this?
```
auto GetLocalPhi(Kokkos::View<double*> Phi_, Kokkos::pair<size_t, size_t> row_range) {
    return Kokkos::subview(Phi_, row_range);
}
```

Reply:
I was thinking the function with the `if` statements for each constraint type could go ahead and calculate the subview of `Phi` and pass it in to the constraint calculation function. Same could be done with the base and target node blocks. It keeps the code from being repeated and should keep the constraint functions cleaner.

Reply:
Alternative idea to achieve the same thing: make Phi into a View with dimensions NumConstraints x 6, similar to the gradient_terms matrix.  We could then just assign into Phi by doing `Phi_(i_constraint, i) = values` without the need for subviews.  The logic of copying to the correct spot in our matrix using row_range would then be moved to `AssembleConstraintsResidual`, just like for the system residual.  We could do something similar by splitting `gradient_terms` into `target_gradient_terms` and `base_gradient_terms`.

These changes would remove our dependence on the ranges in the physics kernels (a good thing) and move them to our Assembly where they belong.  I want to avoid passing subviews of multidimensional views around, when possible, since we then have to either template on their type or use the less efficient (and more verbose) `LayoutStride`.

Reply:
That sounds like a good solution. The constraints residual won't ever have more than 6 items in it and that keeps it consistent with the gradient blocks.